codalab/apps/web/tasks.py

```diff
@@ -605,7 +608,99 @@ def __init__(self, submission, inner_exception):
 
 
 @app.task(queue='submission-updates')
-def update_submission(job_id, args, secret):
+def register_worker(worker_id, ip, cpu_count, mem_mb, harddrive_gb, gpus, vhost):
+    logger.info("Registering worker id = {}, ip = {}, cpu_count = {}, mem_mb = {}, harddrive_gb = {}, gpus = {}, queue = {}".format(
+        worker_id,
+        ip,
+        cpu_count,
+        mem_mb,
+        harddrive_gb,
+        gpus,
+        vhost,
+    ))
+
+    if vhost != '/':
+        queue = Queue.objects.get(vhost=vhost)
+    else:
+        queue = None
+
+    worker = Worker.objects.update_or_create(
+        unique_id=worker_id,
+        defaults={
+            "ip": ip,
+            "cpu_count": cpu_count,
+            "mem_mb": mem_mb,
+            "harddrive_gb": harddrive_gb,
+            "gpus": gpus,
+            "queue": queue
+        },
+    )
+
+
+@app.task(queue='submission-updates')
+def worker_job_started(worker_id, submission_secret, is_scoring):
+    assert worker_id, "WORKER_ID cannot be null, received: {}".format(worker_id)
```

Comment:
General stylistic question: Does it make sense to have asserts in business logic? do we want it to actually throw an error? or should this be something like:
```python
if not worker_id:
    logger.warning('WORKER_ID cannot be null...')
    return
```